### ISSUE TITLE
Simple API without oauth

### ISSUE THREAD

**Issue by paulocastellano** [mautic/mautic#1430]
Hi,

Maybe would be great if you had a simple api with single sign for a key, without oauth2.

**Comment by pranavmehta** [comment]
+1

This is a very useful scenario where a back-end server server needs to post leads into Mautic. 

Simple http basic authentication or a token of sorts in the header kind of thing. Oauth 2 resource owner flow could be another option.

**Comment by healthstatus** [comment]
+1

**Comment by jkuchar** [comment]
What is the problem with OAuth?

**Comment by pranavmehta** [comment]
In case you are using OAuth 2, its token expiry duration. 

Because its a back end server that is posting leads, keeping the token updated frequently is a pain. And setting a very long expiry time, defeats the entire purpose. 

Plus a "user" needs to manually get the token by login in and intercepting the requests during the authentication process.

In short, it simplifies things a lot.

In my case, I don't see a need to use OAuth 1a/2 at all. Leads can be manually created by logging in directly into Mautic or via a back end service which doesn't need a user "granting" permissions.

TBH, I'd be really interested to know where OAuth is a valid use case as far as Mautic is concerned (other than manually exporting data from existing system into Mautic).

**Comment by drmmr763** [comment]
I think mautic uses oauth because so many integrations (CRMs and social) require Oauth as their API's authentication. 

I do think it would be nice for Mautic to support a personal never-expiring auth key that doesn't require the whole log in / exchanging of keys flow so making requests would be a bit easier. 

Oauth is a major part of many APIs (twitter, google, pretty much any popular service these days), so you may want to get used to how it works ;)

**Comment by jkuchar** [comment]
I also do not see point of having something much less secure. When implementing OAuth there are tons of libraries that will make implementation easy.

**Comment by healthstatus** [comment]
Just because you don't see the point doesn't mean there isn't one.  I have a load balanced server and maintaining oauth keys across servers is a big pain.  Ideally something like this would need to be enabled, so if you want to leave it off you could, and those of us that need/want it, can have it.

**Comment by fashionlearn** [comment]
+ 1

**Comment by fashionlearn** [comment]
+ 1

**Comment by fashionlearn** [comment]
+1

**Comment by rodurma** [comment]
+1

**Comment by ahansson89** [comment]
+1

**Comment by craigvansant** [comment]
+1

**Comment by guestisp** [comment]
You can use the "password" or "client_credential"  grant type to act like a standard auth without tokens and still having oauth. 

In fact, client credential workflow is the suggested oauth grant type for autenticating a remote trusted application. There is no need to remove oauth

Obviously the remote application must be trusted and safe.

**Comment by escopecz** [comment]
Guys, take a look at https://github.com/mautic/mautic/pull/1114 if it solves your problem and possibly, leave there a note about your tests.

**Comment by paulocastellano** [comment]
I don't know what problem in "Can send leads to mautic by any **SIMPLE API KEY**", really...

**Comment by ricsmo** [comment]
+1 It would be nice to add leads like this https://medium.com/@jan_linhart/the-simplest-way-how-to-submit-a-form-data-to-mautic-1454d3afd005#.z29r43of5 but without having to create a form on the Mautic side.
Examples:
https://sendy.co/api
https://github.com/twisted1919/mailwizz-php-sdk

FYI I was deciding on whether to purchase sendy or Mailwizz when I found Mautic. I prefer Mautic for everything except having to use oauth for the api.

**Comment by escopecz** [comment]
I like the idea behind [JWT](https://jwt.io/). It will be more secure than one forever-lasting token. Although I've never implemented it yet. Anyone with an experience here?

**Comment by ricsmo** [comment]
I played with it for a short while while trying to do something in node, but didn't get that far because I went in another direction with the project.
I do appreciate the desire for extra security in lead modification, but you do already allow opening up all fields to update via the pixel so some consistency with that and the API would be nice. I want to note that I did prefer the MailWizz SDK method because I feared that others could post fake lead data to my list until I came across a Sendy forum thread where the Sendy team basically said that just about every auto responder provider takes form data anyway so why make it more complicated with extra security when a door is already open? Even Mautic takes post data and this can be used like in the code I linked to. It seems that the front door is overprotected and the back door can be left unlocked.

**Comment by escopecz** [comment]
That is true when your only concern is to push leads to Mautic. Mautic API lets you do much more stuff with data which shouldn't be accessible/editable by 3rd party without permission or easy to hack into. So yes, if you want to push leads without OAuth, there are 2 ways for doing so. But the Mautic API should be secure.

These discussions are more close to what you want I think: https://github.com/mautic/mautic/issues/967#issuecomment-200896125, https://github.com/mautic/mautic/issues/18#issuecomment-199963792

**Comment by ricsmo** [comment]
I think I have a solution for my use case, but it seems hacky. I run training sites and am really impressed that I can use Mautic to manage my emails based on student progress. I'd prefer sending updates via PHP code or post data, but it seems I can also do everything I want using the pixel and tags with campaigns. I'm probably getting a bit off topic, but I wrote my own Wordpress logged in modifications to your plugin until I found your PR that already had it. So I think I have everything I need except that there isn't an easy way for WP plugin developers to integrate with Mautic. It would be nice to see all of the popular popup plugins supporting Mautic like they are starting to for Sendy or MailWizz. But to do that I'm assuming that a simpler lead adding API should exist.
